### PR TITLE
[Enhancements] instructor can always review complaints

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

Some instructors had issues with the way complaints are handled, and there was a bug concerning the second correction round in the complaint list.

### Description

Instructors could not see the complaints of assessments which they assessed. Now, even if the instructor assessed the latest result, he can see the complaint in the exercise-assessment dashboard. For tutors everything works as before.

The "show complaint" button in did not navigate to the correct route, as the correctionRound parameter was not set.
image

### Steps for Testing

1. In  an exam assess a last result (depending if second correction is enabled in the first or second round). 
1.1 Now complain with the user the submission with the assessment belongs to. This instructor user should see the complaint in the exercise-assessment-dashboard. (and all others too)
1.2 With a not-instructor user do the same, and check that after a student complained you cannot see the complaint in the exercise-assessment dashboard.
2. with the instructor go to a course, that has a non exam exercise with complaints. 
2.1 Go to the instructor dashboard and then to complaints. Check that the buttons work as they should.

2.2 Now check for two exams: one with one, and one with two correction-rounds: In the assessment dashboard click on "Total Complaints", in this list the "show complaint" buttons should bring you to the correct

### CODE REVIEW

**Review by ivanchimeno — DISMISSED** [review]
The code looks fine to me and we tested in the testing session

**Review by iozbeyli — APPROVED** [review]
Was in test session, works fine.

File: src/main/webapp/app/complaints/list-of-complaints/list-of-complaints.component.ts [hunk 1]
```diff
@@ -107,21 +108,24 @@ export class ListOfComplaintsComponent implements OnInit {
         const studentParticipation = complaint.result.participation as StudentParticipation;
         const exercise = studentParticipation.exercise;
         const submissionId = complaint.result.submission.id;
-
+        console.log(exercise);
```
Comment (Strohgelaender) [inline src/main/webapp/app/complaints/list-of-complaints/list-of-complaints.component.ts:111]:
```suggestion
```
I think this is just a leftover from debugging.

**Review by nicolasruscher — COMMENTED** [review]
apart from the console log, code is looking good, will approve once log is removed

File: src/main/webapp/app/complaints/list-of-complaints/list-of-complaints.component.ts [hunk 2]
```diff
@@ -107,21 +108,23 @@ export class ListOfComplaintsComponent implements OnInit {
         const studentParticipation = complaint.result.participation as StudentParticipation;
         const exercise = studentParticipation.exercise;
         const submissionId = complaint.result.submission.id;
-
         if (!exercise || !exercise.type || !submissionId) {
             return;
         }
+        if (!this.correctionRound) {
+            this.correctionRound = 0;
+        }
```
Comment (FuchsDominik) [inline src/main/webapp/app/complaints/list-of-complaints/list-of-complaints.component.ts:116]:
```suggestion
        this.correctionRound ?? 0;
```

**Review by FuchsDominik — COMMENTED** [review]
Code looks good

File: src/main/webapp/app/complaints/list-of-complaints/list-of-complaints.component.ts (hunk 2 above)
Comment (SimonEntholzer) [reply src/main/webapp/app/complaints/list-of-complaints/list-of-complaints.component.ts:116]: I guess that would be more concise, but also less readable so I'll keep it. :)

**Review by derLalla — DISMISSED** [review]
code looks also good 👍

**Review by nicolasruscher — DISMISSED** [review]
lgtm 👍

**Review by krusche — COMMENTED** [review]
the code changes look good to me 👍

**Review by iozbeyli — APPROVED** [review]
Code looks good and works fine

**Review by krusche — APPROVED** [review]

**Pull request merged**